src/commands/api-mesh/log-get-bulk.js

```diff
@@ -0,0 +1,208 @@
+
+			this.error(
+				`Max duration between startTime and endTime should be 30 minutes. Current duration is ${hours} hour${
+					hours !== 1 ? 's' : ''
+				} ${minutes} minute${minutes !== 1 ? 's' : ''} and ${seconds} second${
+					seconds !== 1 ? 's' : ''
+				}.`,
+			);
+			return;
+		}
+
+		const { imsOrgId, imsOrgCode, projectId, workspaceId, workspaceName } = await initSdk({
+			ignoreCache,
+		});
+
+		// Retrieve meshId
+		let meshId = null;
+		try {
+			meshId = await getMeshId(imsOrgId, projectId, workspaceId, workspaceName);
+		} catch (err) {
+			this.error(`Unable to get mesh ID: ${err.message}`);
+		}
+
+		if (!meshId) {
+			this.error('Mesh ID not found.');
+		}
+
+		// 5. Call downloadFiles
+		const { presignedUrls, totalSize } = await downloadFilesSequentially(
+			imsOrgCode,
+			projectId,
+			workspaceId,
+			meshId,
+			formattedStartTime,
+			formattedEndTime,
+		);
+		//If presigned URLs are not found, throw error saying that no logs are found
+		if (!presignedUrls || presignedUrls.length === 0) {
+			this.error('No logs found for the given time range.');
+		}
+
+		let shouldDownload = false;
+		if (totalSize > 0) {
+			const totalSizeKB = (totalSize / 1024).toFixed(2); // Convert bytes to KB
+			// 7. Get user confirmation
+			shouldDownload = await promptConfirm(
+				`Expected file size is ${totalSizeKB} KB. Please confirm to output to file ${filename} (y/n)`,
+			);
+			if (shouldDownload) {
+				//create a writer and proceed with download
+				const writer = fs.createWriteStream(outputFile, { flags: 'a' });
+
+				// Write the column headers before appending the log content
+				writer.write(`${columnHeaders}\n`);
+
+				// Stream the data from the signed URLs
+				for (const urlObj of presignedUrls) {
+					const { key, url } = urlObj;
+					logger.info(`Downloading ${key} and appending to ${outputFile}...`);
+
+					try {
+						const fileContentStream = await this.downloadFileContent(url);
+						fileContentStream.pipe(writer, { end: false });
+
+						await new Promise((resolve, reject) => {
+							fileContentStream.on('end', resolve);
+							fileContentStream.on('error', reject);
+						});
+						// write a newline after each file write
+						writer.write('\n');
+
+						logger.info(`${key} content appended successfully.`);
+					} catch (error) {
+						logger.error(`Error downloading or appending content of ${key}:`, error);
+					}
+				}
+
+				this.log(`Successfully downloaded the logs to ${filename}`);
+			} else {
+				this.log('Log files are not downloaded.');
+			}
+		} else {
+			this.error('No logs to download');
+		}
+	}
+
+	async downloadFileContent(url) {
+		return axios({
+			method: 'get',
+			url: url,
+			responseType: 'stream',
+		})
+			.then(response => response.data)
+			.catch(error => {
+				logger.error('Error downloading log content:', error.message);
+				throw error;
+			});
+	}
+}
+
+GetBulkLogCommand.description = 'Download logs in bulk';
```

Comment:
Recommend making this a bit more description. Something like `Download mesh logs for a custom time period`.

cc: @jhadobe